OpenVote/src/com/openvote/LoginServlet.java

```diff
@@ -29,7 +31,14 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
 	
     public void doPost(HttpServletRequest req, HttpServletResponse resp)
             throws IOException {
-    	
+
+
+		UserService userService=UserServiceFactory.getUserService();
+		User user=userService.getCurrentUser();
+
+		//user logged in and is Admin
+		if (user!= null && isAdminLoggedIn()) {
```

Comment:
Okay, I can understand the try/catch's usefulness. I guess the part that makes me feel weird is that we get the user service and user, then call this function that also gets the user cervice and user. Can the user just be passed in? If not, could we just use the function to get the user (in the try/catch) and return null if it's not the admin, or a User type if it is?